src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -950,8 +991,11 @@ internal IEnumerable<SecurityKey> GetContentEncryptionKeys(JsonWebToken jwtToken
             else
             {
                 var key = ResolveTokenDecryptionKey(jwtToken.EncodedToken, jwtToken, validationParameters);
+                if (key == null && configuration != null)
+                    key = ResolveTokenDecryptionKeyFromConfig(jwtToken, configuration);
+
```

Comment:
Seems like we should have some testing to ensure the order of override remains the same e.g.

Prefer delegate override, then static TVP, then metadata.

Is there a reason we don't accept TVP set and metadata set? i.e. why not have both in the collection?